bgpd/bgp_nexthop.c

```diff
@@ -190,9 +190,22 @@ static void show_address_entry(struct hash_bucket *bucket, void *args)
 	struct bgp_addr *addr = (struct bgp_addr *)bucket->data;
 	char *name;
 	struct listnode *node;
+	char str[INET6_ADDRSTRLEN] = {0};
 
-	vty_out(vty, "addr: %s, count: %d : ", inet_ntoa(addr->addr),
-		addr->ifp_name_list->count);
+	if (addr->p->family == AF_INET) {
+
+		vty_out(vty, "addr: %s, count: %d : ",
+				inet_ntoa(addr->p->u.prefix4),
+				addr->ifp_name_list->count);
```

Comment:
dont use inet_ntoa, I think it allocates and never frees a buffer.
please use inet_ntop() instead.